### PR TITLE
Add ECC_SIGNATURE_ALGORITHM env integration test

### PR DESCRIPTION
**Pull request by jacob-delgado**
Add an integration for ECC CA certs. There was a regression a few months back that could have been caught sooner had an integration test existed.

This is based on tests/integration/security/ca_custom_root. Re-factor to follow once this passes tests.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[X] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[X] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by nrjpoddar** [comment]
@jacob-delgado remember to cherry-pick this to older releases 1.6 and 1.7, thanks!

File: tests/integration/security/ca_custom_root/ecc/trust_domain_validation_test.go [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+)
+
+// TestTrustDomainValidation tests the trust domain validation when mTLS is enabled.
+// The trust domain validation should reject a request if it's not from the trust domains configured in the mesh config.
+// The test uses naked client (no sidecar) with custom certificates of different trust domains and covers the following:
+// - plaintext requests are not affected
+// - same trust domain (cluster.local) and aliases (trust-domain-foo and trust-domain-bar)
+// - works for both HTTP and TCP protocol
+// - works for pass through filter chains
+func TestTrustDomainValidation(t *testing.T) {
```
Comment (howardjohn) [inline tests/integration/security/ca_custom_root/ecc/trust_domain_validation_test.go:91]: can we reuse the same test code? i haven't looked at the other test but I imagine this is copy+pasted?

Comment (jacob-delgado) [reply tests/integration/security/ca_custom_root/ecc/trust_domain_validation_test.go:91]: That's the intent. There's much cleanup to do. For now I was trying to make sure getting the env set correctly was working. It looks like I may have to put it in its own top-level test. Is that correct? I'm not sure why there are test failures. I would have thought putting it in its own package would suffice.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**